src/Infrastructure.EntityFramework/AdminConsole/Repositories/OrganizationRepository.cs

```diff
@@ -270,8 +271,199 @@ into grouped
         return await query.ToListAsync();
     }
 
-    public Task EnableCollectionEnhancements(Guid organizationId)
+    public async Task EnableCollectionEnhancements(Guid organizationId)
     {
-        throw new NotImplementedException("Collection enhancements migration is not yet supported for Entity Framework.");
+        using var scope = ServiceScopeFactory.CreateScope();
+        var dbContext = GetDatabaseContext(scope);
+        await using var transaction = await dbContext.Database.BeginTransactionAsync();
+
+        try
+        {
+            // Step 1: AccessAll migration for Groups
+            var groupIdsWithAccessAll = dbContext.Groups
+                .Where(g => g.OrganizationId == organizationId && g.AccessAll == true)
+                .Select(g => g.Id)
+                .ToList();
+
+            // Update existing CollectionGroup rows
+            dbContext.CollectionGroups
+                .Where(cg =>
+                    groupIdsWithAccessAll.Contains(cg.GroupId) &&
+                    cg.Collection.OrganizationId == organizationId)
+                .ToList()
+                .ForEach(cg =>
+                {
+                    cg.ReadOnly = false;
+                    cg.HidePasswords = false;
+                    cg.Manage = false;
+                });
+
+            // Insert new rows into CollectionGroup
+            foreach (var group in groupIdsWithAccessAll)
+            {
+                var newCollectionGroups = dbContext.Collections
+                    .Where(c =>
+                        c.OrganizationId == organizationId &&
+                        !dbContext.CollectionGroups.Any(cg => cg.CollectionId == c.Id && cg.GroupId == group))
```

Comment:
I think this could be

```suggestion
                        !c.CollectionGroups.Any(cg.GroupId == group))
```

Reply:
I ended up changing it to this 
```suggestion
                       c.CollectionGroups.All(cg => cg.GroupId != groupId))
```